core/trino-spi/src/main/java/io/trino/spi/block/Block.java

```diff
@@ -326,4 +326,10 @@ default List<Block> getChildren()
     {
         return Collections.emptyList();
     }
+
+    /**
+     * Returns a block that has an appended null at the end, no matter if the original block has null or not.
+     * The original block won't be modified.
+     */
+    Block newBlockWithAppendedNull();
```

Comment:
`copyWithAppendedNull`?

Reply:
Also if you could calrify in the javadoc that it may not be not supperted and when is that the case.